wallet/src/de/schildbach/wallet/ui/dashpay/HistoryHeaderAdapter.kt

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright 2022 Dash Core Group.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package de.schildbach.wallet.ui.dashpay
+
+import android.content.SharedPreferences
+import android.graphics.drawable.AnimationDrawable
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.view.isVisible
+import androidx.recyclerview.widget.RecyclerView
+import de.schildbach.wallet.data.BlockchainIdentityBaseData
+import de.schildbach.wallet.data.BlockchainIdentityData
+import de.schildbach.wallet_test.R
+import de.schildbach.wallet_test.databinding.HistoryHeaderViewBinding
+
+class HistoryHeaderAdapter(
```

Comment:
DashPay widgets (join, creating identity) are moved into the separate adapter that's used as a header for the transaction history. This allows simpler integration and future merging, as well as provides some separation of concerns.